src/engine.rs

```diff
@@ -169,12 +169,52 @@ pub(crate) async fn start(
                                     // So we need to convert Vec<u8> to pchain_network::Message. Instead of implementing
                                     // TryFrom trait for Vec<u8> to Message, implement a function that takes in the Message Topic to help
                                     // converting Vec<u8> to Message. You can refer to fullnode/mempool messagegate to see how to 
-                                    // deserialise each Message type. 
+                                    // deserialise each Message type.
+
+                                    let topic = config::fullnode_topics(local_public_address)
+                                    .into_iter()
+                                    .find(|t| t.clone().hash() == message.topic);
+
+                                    if let Some(topic) = topic {
+                                        match topic {
+                                            HotStuffRsBroadcast => {
+                                                let hotstuff_message = 
+                                                hotstuff_rs::messages::Message::deserialize(&mut message.data.as_slice())
```

Comment:
What's the reason of getting the de-serialized result of type Result<> and wrap them with Optional and check if it is Some() or None later? Can't we do something directly when the deserialisation gives out Error? 

Also, you gotta return the deserialised message out after this match statement. You should be about to return pchain_network::message::Message.

You might consider getting this part out into a separate helper function to make it look cleaner.